compiler/src/iree/compiler/API2/CMakeLists.txt

```diff
@@ -121,8 +121,12 @@ install(
 
 # The bazel side uses a "Impl" library alias to generically represent
 # static or shared. Provide a corresponding one here.
+# When in BUILD_SHARED_LIBS mode, we do CMake level linking against the
+# non-shared library, since strict layering is enforced there. The mondo
+# shared library can still be generated and used dynamically but is not
+# generally depended on from a normal C++ link.
 
-if(IREE_LINK_COMPILER_SHARED_LIBRARY)
+if(IREE_LINK_COMPILER_SHARED_LIBRARY AND NOT BUILD_SHARED_LIBS)
```

Comment:
Should update the comment below, if you think it's useful. Rather than a comment though, CMake supports an optional argument to `endif` that is enforced to exactly match the argument to `if`. I think that's better in cases where a comment seems necessary (IMO with no nesting and a short if block, it isn't necessary here

Reply:
Removed the comment. This is a small condition and just adding noise.